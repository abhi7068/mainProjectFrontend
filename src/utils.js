import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
