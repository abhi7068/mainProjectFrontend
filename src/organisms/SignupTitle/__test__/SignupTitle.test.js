import React from 'react';
import Signuptitle from '../index';
import { render } from '../../../utils';

it('it renders', () => {
  render(<Signuptitle />);
});
it('should take a snapshot', () => {
  const { asFragment } = render(<Signuptitle />);
  expect(asFragment(render(<Signuptitle />))).toMatchSnapshot();
});
