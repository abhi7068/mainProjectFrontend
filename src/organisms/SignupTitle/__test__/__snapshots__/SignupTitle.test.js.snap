// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-leftTitle-34"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5"
      data-testid="Typography"
      style="font-size: 29px; font-weight: 500;"
    >
      Signup
    </h2>
    <p
      class="makeStyles-subtitle-66"
    >
      <span>
        We do not share your personal details with anyone.
      </span>
    </p>
    <div
      class="makeStyles-positionImage-35"
    >
      <img
        alt="login"
        src="yoyo-logo-mob.png"
        width="200"
      />
    </div>
  </div>
</DocumentFragment>
`;
